src/interface.ts

```diff
@@ -59,11 +69,19 @@ interface Listener {
  */
 interface DesktopAgent {
   /**
-   * Launches/links to an app by name.
+   * Sends the given context to the given target. 
+   * 
+   * Specified observer receives either result object or error object after invokation completed.
+   * For fire-and-forget cases observer can be omited.
+   */
+  invoke(target: Target, context: Context, observer?: InvokationObserver): Cancellable;
```

Comment:
this looks like it should return a Promise rather than using a callback. Also, is this really cancellable? Once you've sent a context to a target, can you cancel the effect it will have?